Basic-Recruit-Training/README.adoc

```diff
@@ -1,8 +1,92 @@
-Basic-Recruit-Training
------------------------
+= Pointers
+
+One of the most important concepts in C/C++ programming is pointers.
+We will look at an example chunk of code that uses pointers to change the value in a variable before exploring pointers more in depth.
+
+----
+int x;
+x = 0;
+int *p;
+p = &x;
+*p = 6;
+
+cout << x << endl; //this prints the number 6
+----
+
+First lets look at a pointer variable.
+
+----
+int * p;
+----
+
+A pointer is a very similar to other primitive data types such as int.
+Instead of containing a value a pointer will instead contain a memory address.
+This address is the location in memory of another variable.
+To initialize a pointer use an asterisk after the type that the pointer will be pointing to.
+This will create a variable p, which is a pointer that will be able to point to a variable of type int.
+In order to assign our pointer p to the address of a variable we will need to use the address operator, &.
+
+----
+int x;
+int *p;
+
+x = 3;
+p = &x;
+----
+
+Now the memory address of x will be stored in p.
+Printing the value of p will print the address of x.
+This will look something like 0x7ffc9b657c7c.
+This means that the value for x is stored on your machine at the memory location 0x7ffc9b657c7c.
+Next lets look at the dereferencing operator.
+
+----
+int x;
+int *p;
+int y;
+
+x = 3;
+p = &x;
+
+y = *p;
+----
+
+Printing the value of y will print 3.
+We use the dereferencing operator to get the value stored at the address pointed to by the pointer.
```

Comment:
We should mention that the address points to the address of the first byte of the object. Also that since this is an int pointer, dereferencing the value gives the value of the 4 bytes it points to.